src/thermo/src/implicit/4C_thermo_timint_impl.cpp

```diff
@@ -88,129 +83,57 @@ Thermo::TimIntImpl::TimIntImpl(const Teuchos::ParameterList& ioparams,
   }
 }
 
-/*----------------------------------------------------------------------*
- | integrate step                                           bborn 08/09 |
- *----------------------------------------------------------------------*/
-void Thermo::TimIntImpl::integrate_step()
-{
-  predict();
-  solve();
-  return;
-}
 
 /*----------------------------------------------------------------------*
- | build linear system tangent matrix, rhs/force residual   bborn 08/09 |
+ | build linear system tangent matrix, rhs/force residual    dano 02/11 |
  | Monolithic TSI accesses the linearised thermo problem                |
  *----------------------------------------------------------------------*/
-void Thermo::TimIntImpl::evaluate(std::shared_ptr<const Core::LinAlg::Vector<double>> tempi)
+void Thermo::TimIntImpl::evaluate()
 {
-  // Yes, this is complicated. But we have to be very careful
-  // here. The field solver always expects an increment only. And
-  // there are Dirichlet conditions that need to be preserved. So take
-  // the sum of increments we get from NOX and apply the latest
-  // increment only.
-  //  if (temp != nullptr)
-  //  {
-  //    // residual temperatures (or iteration increments or iteratively
-  //    // incremental temperatures)
-  //    std::shared_ptr<Core::LinAlg::Vector<double>> tempi = Teuchos::rcp(new
-  //    Core::LinAlg::Vector<double>(*temp)); tempi->Update(-1.0, *tempinc_, 1.0);
-  //
-  //    // update incremental temperature member to provided step increments
-  //    // shortly: tempinc_^<i> := temp^<i+1>
-  //    tempinc_->Update(1.0, *temp, 0.0);
-  //
-  //    // do thermal update with provided residual temperatures
-  //    // recent increment: tempi == tempi_ = \f$\Delta{T}^{<k>}_{n+1}\f$
-  //    thermo_->update_iter_incrementally(tempi);
-  //  }
-  //  else
-  //  {
-  //    thermo_->update_iter_incrementally(nullptr);
-  //  }
-
-  // TSI does not use NOX --> the Newton increment is passed to the field solver
-  update_iter_incrementally(tempi);
-
   // builds tangent, residual and applies DBC
-  evaluate();
+  evaluate_rhs_tang_residual();
+  prepare_system_for_newton_solve();
 }
 
+
 /*----------------------------------------------------------------------*
- | build linear system tangent matrix, rhs/force residual    dano 02/11 |
- | Monolithic TSI accesses the linearised thermo problem                |
  *----------------------------------------------------------------------*/
-void Thermo::TimIntImpl::evaluate()
+void Thermo::TimIntImpl::prepare_time_step()
 {
-  // builds tangent, residual and applies DBC
-  evaluate_rhs_tang_residual();
-  prepare_system_for_newton_solve();
+  predict_step();
+  prepare_step();
```

Comment:
This looks funny. Why `predict` before `prepare`? Maybe the names are just not the best. Probably, it makes sense to further merge these kinds of functions into the algorithm to see what is going on without jumping around so much. `void` functions not taking any parameters often hide too much information, IMO. Just an idea for a next step.